src/utils/sandbox/fetchPatientData.ts

```diff
@@ -1,94 +1,113 @@
+ */
+export const fetchPatientData = async (fhirClient: Client, progress: ProgressMonitor): Promise<PatientData> => {
+  const tasks: FetchTaskPromiseType = [
     fhirClient.patient.read(),
     fhirClient.user.read(),
-    getCondition(MCODE_PRIMARY_CANCER_CONDITION),
-    getCondition(MCODE_SECONDARY_CANCER_CONDITION),
-    getObservation(MCODE_DISEASE_STATUS),
-    getObservation(MCODE_ECOG_PERFORMANCE_STATUS),
-    getObservation(MCODE_KARNOFSKY_PERFORMANCE_STATUS),
-    getObservation(MCODE_TUMOR_MARKER),
-    getProcedure(MCODE_CANCER_RELATED_RADIATION_PROCEDURE),
-    getProcedure(MCODE_CANCER_RELATED_SURGICAL_PROCEDURE),
-    getMedicationStatement(MCODE_CANCER_RELATED_MEDICATION_STATEMENT),
-  ].map(promise =>
-    promise.then(result => {
-      progress(1);
-      return result;
-    })
-  );
+    fetchResources<Condition>(fhirClient, 'Condition'),
+    fetchResources<Observation>(fhirClient, 'Observation'),
+    fetchResources<Procedure>(fhirClient, 'Procedure'),
+    fetchMedications(fhirClient),
+  ];
   progress('Fetching patient data...', 0, tasks.length);
 
-  const [
-    fhirPatient,
-    fhirUser,
-    fhirPrimaryCancerCondition,
-    fhirSecondaryCancerCondition,
-    fhirDiseaseStatus,
-    fhirEcogPerformanceStatus,
-    fhirKarnofskyPerformanceStatus,
-    fhirTumorMarkers,
-    fhirRadiationProcedures,
-    fhirSurgeryProcedures,
-    fhirMedicationStatements,
-  ] = await Promise.all(tasks);
+  return buildPatientData(
+    (await Promise.all(
+      tasks.map(promise =>
+        promise.then(result => {
+          progress(1);
+          return result;
+        })
+      )
+    )) as FetchTaskType
+  );
+};
+
+/**
+ * This method is split out to make testing easier and is not intended to be used directly. Instead it should be called
+ * via fetchPatientData().
+ * @param param0 the loaded data
+ * @returns
+ */
+export const buildPatientData = ([
+  fhirPatient,
+  fhirUser,
+  conditions,
+  observations,
+  procedures,
+  medications,
+]: FetchTaskType): PatientData => {
+  // FIXME: Should find the most recent, which is not necessarily the first record in the bundle
+  // TODO: As this gets more complicated, it'll make more sense to go through all observations and check each one to see
+  // if it contains relavent information rather than do separate find/filters
+  // Find the disease status based on the required LOINC code
+  const diseaseStatusObservation = observations.find(observation =>
+    observationHasCode(observation, LOINC_CODE_URI, MCODE_CANCER_DISEASE_STATUS_LOINC_CODE)
+  );
+  // FIXME: ECOG and Karnofsy have no associated records, so the following will never work but needs to be changed to
+  // not be using the profile anyway
+  const ecogObservation = observations.find(resource => resourceHasProfile(resource, MCODE_ECOG_PERFORMANCE_STATUS));
+  const karnosfkyObservation = observations.find(resource =>
+    resourceHasProfile(resource, MCODE_KARNOFSKY_PERFORMANCE_STATUS)
+  );
```

Comment:
I'm a bit confused on this; while I understand that Patient records at the current moment do not have ECOG/Karnofsky, shouldn't we just go ahead and change this still to not use profile?

Reply:
The current code for pulling Karnosky/ECOG values doesn't check what the Observation is coding for, it just sees if the Observation.interpretation is one of the ECOG/Karnofsky codes it knows about. So it would also be possible to just go through all of the Observation records and return the first one that doesn't return `null` for that check, but really the entire process of how we find Karnosky/ECOG scores should be revisited based on how they'd be stored in "real" patient records.

Reply:
Makes sense! Talked with @zlister and we agreed that we can make a new ticket to address this.